galadriel/agent.py

```diff
@@ -304,6 +364,51 @@ async def _run_request(self, request: Message):
                 except Exception:
                     logger.error("Failed to send response via output", exc_info=True)
 
+    async def _run_request_stream(self, request: Message):
```

Comment:
Duplication of `_run_request` and `_run_request_stream` will be costly and error-prone in maintenance. Consider deduplicating.